src/components/notification/index.tsx

```diff
@@ -107,45 +86,75 @@ export const Notification: React.FC<NotificationProps> = ({
     }
   `;
 
+  const handleEmbeddedUrl = (type: string, data: string) => {
+    new IterableActionRunner().executeAction(
+      null,
+      { type, data },
+      IterableActionSource.EMBEDDED
+    );
+  };
+
   return (
     <>
       <style>{mediaStyle}</style>
-      <div className="notification" style={cardStyle} onClick={onClickView}>
+      <div
+        className="notification"
+        style={cardStyle}
+        onClick={
+          handleEmbeddedClick
+            ? handleEmbeddedClick
+            : () =>
+                handleEmbeddedUrl(
+                  messageData?.defaultAction?.type,
+                  messageData?.defaultAction?.data
+                )
+        }
+      >
         <div style={{ ...defaultTextParentStyles }}>
           <text
             className="titleText"
             style={{ ...defaultTitleStyles, ...titleStyle }}
           >
-            {title}
+            {messageData?.title}
           </text>
           <text
             className="titleText"
             style={{ ...defaultTextStyles, ...textStyle }}
           >
-            {description}
+            {messageData?.body}
           </text>
         </div>
         <div style={notificationButtons}>
           <div style={notificationButtons}>
-            {primaryButtonLabel && (
+            {messageData?.buttons?.[0].title && (
```

Comment:
we should just be able to map over the buttons here. `messageData?.buttons?.map(button =>...`

Reply:
Done